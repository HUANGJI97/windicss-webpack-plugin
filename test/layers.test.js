"use strict";

const getModuleSource = require('./helpers/getModuleSource')
const vueWebpackCompiler = require ('./helpers/vueWebpackCompiler')
const reactWebpackCompiler = require ('./helpers/reactWebpackCompiler')

describe("Layers test", function() {

  it('should render vue', done => {
    const compiler = vueWebpackCompiler('vue-layers')
    compiler.run((err, stats) => {
      expect(stats.compilation.errors).toStrictEqual([])
      // check the windi file has generated the right classes
      const windiBase = getModuleSource('virtual:windi-base.css', stats)
      expect(windiBase).toContain('body {')
      expect(windiBase).toMatchSnapshot('windi base')

      const windiComponents = getModuleSource('virtual:windi-components.css', stats)
      expect(windiComponents).toContain('windicss layer components')
      expect(windiComponents).toMatchSnapshot('windi components')

      const windiUtils = getModuleSource('virtual:windi-utilities.css', stats)
      expect(windiUtils).toContain('bg-teal-900:hover')
      expect(windiUtils).toMatchSnapshot('windi utilities')

      done(err)
    });
  });
  it('should render react', done => {
    const compiler = reactWebpackCompiler('react-layers')
    compiler.run((err, stats) => {
      expect(stats.compilation.errors).toStrictEqual([])
      // check the windi file has generated the right classes
      const windiBase = getModuleSource('virtual:windi-base.css', stats)
      expect(windiBase).toContain('body {')
      expect(windiBase).toMatchSnapshot('windi base')

      const windiComponents = getModuleSource('virtual:windi-components.css', stats)
      expect(windiComponents).toContain('windicss layer components')
      expect(windiComponents).toMatchSnapshot('windi components')

      const windiUtils = getModuleSource('virtual:windi-utilities.css', stats)
      expect(windiUtils).toContain('bg-blue-500:hover')
      expect(windiUtils).toMatchSnapshot('windi utilities')

      done(err)
    });
  });
});
