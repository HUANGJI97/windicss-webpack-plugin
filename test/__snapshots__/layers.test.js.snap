// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform template test should render vue: windi base 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\"/* virtual:windi-base.css(boot) */\\\\n/* windicss layer base */\\\\n*, ::before, ::after {\\\\n  -webkit-box-sizing: border-box;\\\\n  box-sizing: border-box;\\\\n  border-width: 0;\\\\n  border-style: solid;\\\\n  border-color: #e5e7eb;\\\\n}\\\\n* {\\\\n  --tw-ring-inset: var(--tw-empty,/*!*/ /*!*/);\\\\n  --tw-ring-offset-width: 0px;\\\\n  --tw-ring-offset-color: #fff;\\\\n  --tw-ring-color: rgba(59, 130, 246, 0.5);\\\\n  --tw-ring-offset-shadow: 0 0 #0000;\\\\n  --tw-ring-shadow: 0 0 #0000;\\\\n  --tw-shadow: 0 0 #0000;\\\\n}\\\\n:root {\\\\n  -moz-tab-size: 4;\\\\n  -o-tab-size: 4;\\\\n  tab-size: 4;\\\\n}\\\\n:-moz-focusring {\\\\n  outline: 1px dotted ButtonText;\\\\n}\\\\n:-moz-ui-invalid {\\\\n  box-shadow: none;\\\\n}\\\\n::moz-focus-inner {\\\\n  border-style: none;\\\\n  padding: 0;\\\\n}\\\\n::-webkit-inner-spin-button, ::-webkit-outer-spin-button {\\\\n  height: auto;\\\\n}\\\\n::-webkit-search-decoration {\\\\n  -webkit-appearance: none;\\\\n}\\\\n::-webkit-file-upload-button {\\\\n  -webkit-appearance: button;\\\\n  font: inherit;\\\\n}\\\\n[type='search'] {\\\\n  -webkit-appearance: textfield;\\\\n  outline-offset: -2px;\\\\n}\\\\nabbr[title] {\\\\n  -webkit-text-decoration: underline dotted;\\\\n  text-decoration: underline dotted;\\\\n}\\\\nbody {\\\\n  margin: 0;\\\\n  font-family: inherit;\\\\n  line-height: inherit;\\\\n}\\\\nbutton {\\\\n  font-family: inherit;\\\\n  font-size: 100%;\\\\n  line-height: 1.15;\\\\n  margin: 0;\\\\n  text-transform: none;\\\\n  background-color: transparent;\\\\n  background-image: none;\\\\n  padding: 0;\\\\n  line-height: inherit;\\\\n  color: inherit;\\\\n}\\\\nbutton, [type='button'], [type='reset'], [type='submit'] {\\\\n  -webkit-appearance: button;\\\\n}\\\\nbutton:focus {\\\\n  outline: 1px dotted;\\\\n  outline: 5px auto -webkit-focus-ring-color;\\\\n}\\\\nbutton, [role=\\\\\\"button\\\\\\"] {\\\\n  cursor: pointer;\\\\n}\\\\nhtml {\\\\n  -webkit-text-size-adjust: 100%;\\\\n  font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,\\\\\\"Segoe UI\\\\\\",Roboto,\\\\\\"Helvetica Neue\\\\\\",Arial,\\\\\\"Noto Sans\\\\\\",sans-serif,\\\\\\"Apple Color Emoji\\\\\\",\\\\\\"Segoe UI Emoji\\\\\\",\\\\\\"Segoe UI Symbol\\\\\\",\\\\\\"Noto Color Emoji\\\\\\";\\\\n  line-height: 1.5;\\\\n}\\\\nh2 {\\\\n  margin: 0;\\\\n  font-size: inherit;\\\\n  font-weight: inherit;\\\\n}\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;

exports[`Transform template test should render vue: windi components 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\"/* virtual:windi-components.css(boot) */\\\\n/* windicss layer components */\\\\n\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;

exports[`Transform template test should render vue: windi utilities 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\"/* virtual:windi-utilities.css(boot) */\\\\n/* windicss layer utilities */\\\\n.hover\\\\\\\\:bg-teal-900:hover {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(19, 78, 74, var(--tw-bg-opacity));\\\\n}\\\\n.bg-teal-100 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(204, 251, 241, var(--tw-bg-opacity));\\\\n}\\\\n.bg-black {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(0, 0, 0, var(--tw-bg-opacity));\\\\n}\\\\n.bg-blue-200 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(191, 219, 254, var(--tw-bg-opacity));\\\\n}\\\\n.bg-red-200 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(254, 202, 202, var(--tw-bg-opacity));\\\\n}\\\\n.\\\\\\\\!bg-blue-200 {\\\\n  --tw-bg-opacity: 1 !important;\\\\n  background-color: rgba(191, 219, 254, var(--tw-bg-opacity)) !important;\\\\n}\\\\n.bg-gray-100 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(243, 244, 246, var(--tw-bg-opacity));\\\\n}\\\\n.dark .dark\\\\\\\\:bg-gray-900 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(17, 24, 39, var(--tw-bg-opacity));\\\\n}\\\\n.hover\\\\\\\\:bg-blue-500:hover {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(59, 130, 246, var(--tw-bg-opacity));\\\\n}\\\\n.hover\\\\\\\\:rounded-full:hover {\\\\n  border-radius: 9999px;\\\\n}\\\\n.rounded {\\\\n  border-radius: 0.25rem;\\\\n}\\\\n.h-100vh {\\\\n  height: 100vh;\\\\n}\\\\n.m-0 {\\\\n  margin: 0px;\\\\n}\\\\n.overflow-hidden {\\\\n  overflow: hidden;\\\\n}\\\\n.p-1 {\\\\n  padding: 0.25rem;\\\\n}\\\\n.p-0 {\\\\n  padding: 0px;\\\\n}\\\\n.px-5 {\\\\n  padding-left: 1.25rem;\\\\n  padding-right: 1.25rem;\\\\n}\\\\n.hover\\\\\\\\:text-green-100:hover {\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(209, 250, 229, var(--tw-text-opacity));\\\\n}\\\\n.text-gray-900 {\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(17, 24, 39, var(--tw-text-opacity));\\\\n}\\\\n.dark .dark\\\\\\\\:text-gray-100 {\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(243, 244, 246, var(--tw-text-opacity));\\\\n}\\\\n.hover\\\\\\\\:w-1\\\\\\\\/3:hover {\\\\n  width: 33.333333%;\\\\n}\\\\n.w-100vw {\\\\n  width: 100vw;\\\\n}\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;
